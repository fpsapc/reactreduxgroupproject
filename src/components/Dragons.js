import { useDispatch, useSelector } from 'react-redux';
import { reservation } from '../redux/dragons/dragonSlice';
import '../styles/dragons.css';

function Dragons() {
  const dispatch = useDispatch();
  const { dragons, isLoading, error } = useSelector((store) => store.dragons);

  if (isLoading) {
    return (
      <>
        <div>Loading...</div>
      </>
    );
  }

  if (error) {
    return (
      <>
        <div>Error: Connection is bad!</div>
      </>
    );
  }

  return (
    <>
      <div className="dragons">
        {dragons.map((dragon) => (
          <>
            <div key={dragon.id} className="dragon-container">
              <div className="dragon-img-container">
                <img className="dragon-img" src={dragon.image[0]} alt="dragon view" />
              </div>
              <div className="dragon-info">
                <p className="dragon-name">{dragon.name}</p>
                <p className="dragon-type">
                  {dragon.reserved ? <span className="reserved">Reserved</span> : ''}
                  {dragon.type}
                </p>
                <button className={dragon.reserved ? 'cancel' : 'dragon-button'} id="btn" onClick={() => dispatch(reservation(dragon.id))} type="button">
                  {
                    dragon.reserved
                      ? 'Cancel Reservation'
                      : 'Reserve'
                  }
                </button>
              </div>
            </div>
          </>
        ))}
      </div>
    </>
  );
}

export default Dragons;
