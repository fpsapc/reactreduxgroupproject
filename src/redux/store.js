import { configureStore } from '@reduxjs/toolkit';
import missionReducer from './mission/missionSlice';
import dragonReducer, { fetchDragons } from './dragons/dragonSlice';
import rocketReducer, { fetchRockets } from './rockets/rocketsSlice';

const store = configureStore({
  reducer: {
    rockets: rocketReducer,
    dragons: dragonReducer,
    missions: missionReducer,
  },
});

store.dispatch(fetchRockets());
store.dispatch(fetchDragons());

export default store;
