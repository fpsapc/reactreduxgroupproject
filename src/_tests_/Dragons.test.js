import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../redux/store';
import Dragons from '../components/Dragons';

test('render correctly and matches the Dragons snapshot', () => {
  const { asFragment } = render(
    <Provider store={store}>
      <Dragons />
    </Provider>,
  );
  expect(asFragment()).toMatchSnapshot();
});
