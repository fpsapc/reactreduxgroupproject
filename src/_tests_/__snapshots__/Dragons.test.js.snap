// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly and matches the Dragons snapshot 1`] = `
<DocumentFragment>
  <div>
    Loading...
  </div>
</DocumentFragment>
`;
